bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/api/IntesisBoxMessage.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.intesis.internal.api;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * @author Cody Cutrer - Initial contribution
+ */
+@NonNullByDefault
+public class IntesisBoxMessage {
+    public static final String ID = "ID";
+    public static final String INFO = "INFO";
+    public static final String SET = "SET";
+    public static final String CHN = "CHN";
+    public static final String GET = "GET";
+    public static final String LOGIN = "LOGIN";
+    public static final String LOGOUT = "LOGOUT";
+    public static final String CFG = "CFG";
+    public static final String LIMITS = "LIMITS";
+    public static final String DISCOVER = "DISCOVER";
+
+    private static final Pattern REGEX = Pattern.compile(
+            "^(ID|INFO|SET|CHN|GET|LOGIN|LOGOUT|CFG|LIMITS)(?:,(\\d+))?:(APPVERSION|RUNVERSION|CFGVERSION|HASH|ONOFF|MODE|SETPTEMP|FANSP|VANEUD|VANELR|AMBTEMP|ERRSTATUS|ERRCODE),([A-Z0-9.,\\[\\]]+)$");
```

Comment:
It would be better to parse out the command and function names in their entirety then check that the parsed values are acceptable. That way when you find a mismatch you can log the unexpected value vs not knowing what went wrong.
```suggestion
            "^([^,]+)(?:,(\\d+))?:([^,]+),([A-Z0-9.,\\[\\]]+)$");
```

Reply:
Done